plugins/browser-plugin-button-click-tracking/src/api.ts

```diff
@@ -0,0 +1,104 @@
+
+import { createEventFromButton, filterFunctionFromFilter } from './util';
+import { buildButtonClick } from './core';
+import { ButtonClickEvent, ButtonClickTrackingConfiguration, Config, FilterFunction } from './types';
+
+const _trackers: Record<string, BrowserTracker> = {};
+const _config: Record<string, Config> = {};
+
+/**
+ * Button click tracking
+ *
+ * Will automatically tracking button clicks once enabled with 'enableButtonClickTracking'
+ * or you can manually track button clicks with 'trackButtonClick'
+ */
+export function ButtonClickTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+  };
+}
+
+/**
+ * Manually log a click
+ *
+ * @param event - The event information
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackButtonClick(
+  event: ButtonClickEvent & CommonEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildButtonClick(event), event.context, event.timestamp);
+  });
+}
+
+/**
+ * Enable automatic click tracking for all `<button>` and `<input type="button">` elements on the page
+ *
+ * @param configuration  - The configuration for automatic button click tracking
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function enableButtonClickTracking(
+  configuration: ButtonClickTrackingConfiguration = {},
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  trackers.forEach((trackerId) => {
+    // Store the configuration for this tracker, if it doesn't already exist
+    // This allows us to enable click tracking for a tracker if it has been disabled
+    if (!_config[trackerId]) {
+      const filter = filterFunctionFromFilter(configuration.filter);
+      _config[trackerId] = {
+        filter,
+        context: configuration.context,
+        listener: (event: MouseEvent) => eventHandler(event, trackerId, filter, configuration.context),
+      };
+    }
+
+    const addClickListener = () => {
+      document.addEventListener('click', _config[trackerId].listener);
+    };
+
+    if (_trackers[trackerId]?.sharedState.hasLoaded) {
+      // the load event has already fired, add the click listeners now
+      addClickListener();
+    } else {
+      // defer until page has loaded
+      _trackers[trackerId]?.sharedState.registeredOnLoadHandlers.push(addClickListener);
+    }
+  });
+}
+
+/**
+ * Disable automatic click tracking for all `<button>` and `<input type="button">` elements on the page
+ *
+ * Can be re-enabled with {@link enableButtonClickTracking}
+ */
+export function disableButtonClickTracking() {
+  for (const trackerId in _trackers) {
+    document.removeEventListener('click', _config[trackerId].listener);
+  }
```

Comment:
Can we also clear the `_config`? Just in case someone tries calling the method twice so that it doesn't try to remove the listener twice.